scanpy/get.py

```diff
@@ -295,3 +396,150 @@ def _set_obs_rep(adata, val, *, use_raw=False, layer=None, obsm=None, obsp=None)
+        groupby is a categorical. If groupby is not a categorical observation,
+        it would be subdivided into `num_categories`.
+    use_raw
+        Use `raw` attribute of `adata` if present.
+    log
+        Use the log of the values
+    num_categories
+        Only used if groupby observation is not categorical. This value
+        determines the number of groups into which the groupby observation
+        should be subdivided.
+    gene_symbols
+        Key for field in .var that stores gene symbols.
+    concat_indices
+        Concatenates categorical indices into a single categorical index, if 
+        groupby is a sequence. True by default.
+
+    Returns
+    -------
+    Tuple of `pandas.DataFrame` and list of categories.
+    """
+    from scipy.sparse import issparse
+
+    sanitize_anndata(adata)
+    if use_raw is None and adata.raw is not None:
+        use_raw = True
+    if isinstance(var_names, str):
+        var_names = [var_names]
+    if var_names is None:
+        if use_raw:
+            var_names = adata.raw.var_names.values
+        else:
+            var_names = adata.var_names.values
+
+    if groupby is not None:
+        if isinstance(groupby, str):
+            # if not a list, turn into a list
+            groupby = [groupby]
+        for group in groupby:
+            if group not in adata.obs_keys():
+                raise ValueError(
+                    'groupby has to be a valid observation. '
+                    f'Given {group}, is not in observations: {adata.obs_keys()}'
+                )
+
+    if gene_symbols is not None and gene_symbols in adata.var.columns:
+        # translate gene_symbols to var_names
+        # slow method but gives a meaningful error if no gene symbol is found:
+        translated_var_names = []
+        # if we're using raw to plot, we should also do gene symbol translations
+        # using raw
+        if use_raw:
+            adata_or_raw = adata.raw
+        else:
+            adata_or_raw = adata
+        for symbol in var_names:
+            if symbol not in adata_or_raw.var[gene_symbols].values:
+                logg.error(
+                    f"Gene symbol {symbol!r} not found in given "
+                    f"gene_symbols column: {gene_symbols!r}"
+                )
+                return
+            translated_var_names.append(
+                adata_or_raw.var[adata_or_raw.var[gene_symbols] == symbol].index[0]
+            )
+        symbols = var_names
+        var_names = translated_var_names
+    if layer is not None:
+        if layer not in adata.layers.keys():
+            raise KeyError(
+                f'Selected layer: {layer} is not in the layers list. '
+                f'The list of valid layers is: {adata.layers.keys()}'
+            )
+        matrix = adata[:, var_names].layers[layer]
+    elif use_raw:
+        matrix = adata.raw[:, var_names].X
+    else:
+        matrix = adata[:, var_names].X
```

Comment:
We've got a function for this:

```suggestion
    matrix = sc.get._get_obs_rep(adata[:, var_names], use_raw=use_raw, layers=layer)
```